### PR TITLE
Project monitoring fix

### PR DESCRIPTION
**Pull request by orangedeng**
Base on PR: https://github.com/rancher/system-charts/pull/36

This PR contains following changes
- Remove RBAC resource from project monitoring charts
- Use the service account injected by Rancher in project monitoring charts

Related issue: https://github.com/rancher/rancher/issues/19381

### CODE REVIEW

**Comment by thxCode** [comment]
LGTM

**Comment by guangbochen** [comment]
LGTM, just small lint failures.
```
charts/rancher-monitoring/v0.0.3/values.yaml
  35:9      error    trailing spaces  (trailing-spaces)
  49:9      error    trailing spaces  (trailing-spaces)
  63:9      error    trailing spaces  (trailing-spaces)
  83:9      error    trailing spaces  (trailing-spaces)
  100:9     error    trailing spaces  (trailing-spaces)
  123:9     error    trailing spaces  (trailing-spaces)
  135:9     error    trailing spaces  (trailing-spaces)
```

**Comment by orangedeng** [comment]
Can we have drone for system-charts? @loganhz

**Comment by orangedeng** [comment]
Updated

File: charts/rancher-monitoring/v0.0.3/charts/prometheus/templates/prometheus.yaml [hunk 1]
```diff
@@ -128,13 +133,10 @@ spec:
 {{ toYaml .Values.secrets | indent 2 }}
 {{- end }}
 {{- if .Values.enabledRBAC }}
-  serviceAccountName: {{ default (default (include "app.fullname" .) .Values.serviceAccountName) .Values.serviceAccountNameOverride }}
-{{- end }}
-{{- if .Values.serviceMonitorNamespaceSelector }}
-  serviceMonitorNamespaceSelector:
-{{ toYaml .Values.serviceMonitorNamespaceSelector | indent 4 }}
+  serviceAccountName: {{ default (default (include "app.fullname" .) .Values.serviceAccountNameOverride) .Values.global.serviceAccountName }}
```
Comment (cjellick) [inline charts/rancher-monitoring/v0.0.3/charts/prometheus/templates/prometheus.yaml]: what does `default (default ...` mean or do?

File: charts/rancher-monitoring/v0.0.3/charts/prometheus/templates/_helpers.tpl [hunk 2]
```diff
@@ -56,3 +56,51 @@
 {{- $name := include "app.name" . -}}
 {{- printf "%s-auth-%s" $name .Release.Name -}}
 {{- end -}}
+
```
Comment (cjellick) [inline charts/rancher-monitoring/v0.0.3/charts/prometheus/templates/_helpers.tpl:59]: what do these changes do?

Comment (orangedeng) [reply charts/rancher-monitoring/v0.0.3/charts/prometheus/templates/_helpers.tpl:59]: This is to make sure we set namespace selector for rule and service monitor when installing project level monitoring.

**Comment by guangbochen** [comment]
LGTM

**Pull request merged**